packages/ic-asset-certification/src/asset_router.rs

```diff
@@ -315,6 +312,59 @@ impl<'content> AssetRouter<'content> {
         Ok(())
     }
 
+    /// Deletes multiple assets from the router, including any certification for those assets.
+    ///
+    /// Depending on the configuration provided to the [certify_assets](AssetRouter::certify_assets) function,
+    /// multiple responses may be generated for the same asset. To ensure that all generated responses are deleted,
+    /// this function accepts the same configuration.
+    pub fn delete_assets<'path>(
```

Comment:
`delete_asset` very closely replicates the logic from `certify_assets`, but deletes assets instead of certifies them.

Likewise, `delete_asset_impl` closely reflects `certify_asset_impl`.

There are other pairs too:
- `insert_static_asset` / `delete_static_asset`
- `insert_fallback_asset` / `delete_fallback_asset`
- `insert_redirect` / `delete_redirect`

Those pairs had their common code factored out into the following functions:
- `prepare_static_asset`
- `prepare_fallback_asset`
- `prepare_redirect`